### PR TITLE
ops(release): PPO baseline training launcher for #371 frozen baselines

### LINKED ISSUE #384: ops(release): launch PPO baseline training on alcubierre cluster for #371 frozen baselines
**Parent**: #371 (Frozen baseline release, slice 3/5 of #365)

## Why filed

#371's acceptance includes "Trained PPO baseline checkpoints per scenario (from M2.1 / #360) with reported scores in a \`scores.json\`". #360 is currently \`loom:blocked\` because the upstream Nash phase diagram (#358 → now Run 2a in #381) hasn't produced its results. But the *operator-driven* part — actually running PPO on the alcubierre cluster — is a concrete launchable task that doesn't require waiting on the full #360 ramp. This issue captures that subset for execution as soon as a stable training config is available.

## Goal

Run PPO baseline training across the canonical scenario set on the alcubierre cluster (\`\$COMPUTE_HOST_CLUSTER\`), producing per-scenario checkpoints and a \`scores.json\` ready to be wrapped by #371's loader API.

## Scenarios to train (from the canonical scenario registry shipped in slice 1 / #369)

Reference: \`bucket_brigade/envs/scenarios_generated.py\` and its \`_SCENARIO_REGISTRY\` dict. Suggested initial set:

- \`minimal_specialization-v1\` (canonical P3 substrate)
- \`default-v1\`
- \`positional_default-v1\`
- \`chain_reaction-v1\`
- \`trivial_cooperation-v1\`
- \`v2_minimal-v1\` (2 houses × 4 agents)

Skip scenarios the paper doesn't reference in headline numbers — verify the final list against the paper outline before launching.

## Operator workflow (manual — \`loom:curated\`, not \`loom:issue\`)

1. \`source .env && ssh \"\$COMPUTE_HOST_CLUSTER\"\` (alcubierre)
2. \`cd ~/GitHub/bucket-brigade && git pull && bash bucket-brigade-core/build.sh\`
3. \`tmux new -s ppo-baselines\`
4. Per scenario, launch a training run with the project's standard PPO config. Reference command (verify exact flags against \`experiments/p3_specialization/train.py\` argparse before running):
   \`\`\`bash
   uv run python -m experiments.p3_specialization.train \\
     --scenario \$SCENARIO --seeds 42 43 44 \\
     --num-iterations 50 --output-root experiments/p3_specialization/baselines/\$SCENARIO/ \\
     2>&1 | tee logs/ppo_\${SCENARIO}_\$(date +%Y%m%d_%H%M%S).log
   \`\`\`
5. Detach + monitor periodically. Estimated wall clock per scenario: ~30 min – 2h GPU-time × 3 seeds (see #360 compute estimate).
6. On completion, package the best checkpoint per scenario into \`bucket_brigade/baselines/release/ppo/\$SCENARIO/checkpoint.pt\` with a sibling \`config.json\` capturing training hyperparameters and gap_closed score.
7. Author \`bucket_brigade/baselines/release/ppo/scores.json\` summarizing per-scenario {seed, gap_closed, config_hash}.
8. Open a results-only PR with \`loom:review-requested\` and \`Refs #371\` (don't close #371 — that's bigger than this run).

## Compute estimate

- 6 scenarios × 3 seeds × ~1h average = ~18 GPU-hours on alcubierre

## Acceptance criteria

- [ ] Per-scenario checkpoint in \`bucket_brigade/baselines/release/ppo/\$SCENARIO/checkpoint.pt\`
- [ ] \`scores.json\` lists every trained scenario with gap_closed value + training config hash
- [ ] Loader API (to be added by #371) can deserialize and run the checkpoints in a smoke test
- [ ] Results-only PR merged; #371 Acceptance "trained PPO baseline checkpoints" box ticked

## Out of scope

- Loader API itself (that lives in #371; this issue only produces the artifacts)
- HuggingFace upload (separate slice — #373)
- Full #360 trainability sweep across the NE phase diagram (this is the *baseline* checkpoints for the release, not the full sweep — those are separate runs once #381 Run 2 results land)

## References

- Parent: #371
- Sibling upstream (blocked): #360 (full PPO trainability sweep)
- Slice-1 dependency (closed, merged): #369 (env API + scenario registry)
- Scenario registry source: \`bucket_brigade/envs/scenarios_generated.py:_SCENARIO_REGISTRY\`
- Training script: \`experiments/p3_specialization/train.py\`

## Why filed now

Surfaced during the 2026-06-05 sweep. User has alcubierre cluster access; making the PPO baseline run a discrete scheduleable item turns #371 from "blocked indefinitely on #360" into "blocked on operator scheduling cluster time + Nash results landing".

### PR DESCRIPTION
**Pull request by rjwalters**
## Summary

Adds the operator-launch wrapper for the PPO baseline training sweep that produces the frozen checkpoints shipped with the M4 release. Closes #384, feeds parent #371 (slice 3/5 of #365).

This is the fourth Option-A launcher in this series (alongside #393's phase-diagram fill, #394's Tier-1 sweep, #395's het_ppo arm). Same pattern, same shape — launcher + runbook + contract tests.

### What lands

- `experiments/scripts/launch_ppo_baselines.sh` — operator-launch wrapper that SSHs into a `$COMPUTE_HOST_*` host, builds the Rust extension, and starts a detached tmux session running `run_tier1_cell.py --trainer ippo` over the canonical #384 release scenario set
- `experiments/p3_specialization/PPO_BASELINES_RUNBOOK.md` — operator runbook with 4 launch plans (full sweep / sanity / shard / smoke) and the handoff procedure to #371
- `tests/test_launch_ppo_baselines.py` — 21 contract tests (arg parsing, host resolution, driver-command synthesis, session-name compaction, runbook canonical invocations, drift check against the driver's TRAINERS dict)

### What this does NOT do

- Does not run training locally (CLAUDE.md is explicit on this)
- Does not package checkpoints into `bucket_brigade/baselines/release/ppo/` — that is #371's job
- Does not auto-launch — operator runs `./experiments/scripts/launch_ppo_baselines.sh` to kick off the remote sweep

### Default scenario set (per #384)

`minimal_specialization`, `default`, `positional_default`, `chain_reaction`, `trivial_cooperation`, `v2_minimal` × seeds 42,43,44 × 50 iter × 2048 rollout = ~18 h on `$COMPUTE_HOST_CLUSTER` per the #384 estimate.

### Caveats

- The issue body's "Reference command" calls `train.py --seeds 42 43 44`, but `train.py` takes a single `--seed`. The launcher uses `run_tier1_cell.py --trainer ippo` instead — the multi-seed wrapper that dispatches to train.py per-seed, matching the sibling launchers (#394, #395). Documented in the runbook.
- `positional_default` is in `SCENARIO_REGISTRY` (legacy name-keyed) but not in `SCENARIO_VERSIONS` (versioned IDs). Training works fine; #371's loader may need a `positional_default-v1` entry added — flagged in the runbook.

## Test plan

- [x] `uv run pytest tests/test_launch_ppo_baselines.py` — 21 passed
- [x] `uv run pytest tests/test_launch_*.py` — 63 passed (no regressions on sibling launchers)
- [x] `uv run ruff check tests/test_launch_ppo_baselines.py` — clean
- [x] `uv run ruff format --check tests/test_launch_ppo_baselines.py` — clean
- [x] `bash experiments/scripts/launch_ppo_baselines.sh --dry-run --host fake-host --skip-connectivity-check` — produces the expected 6-cell plan
- [ ] Operator smoke (out of scope for this PR): `bash launch_ppo_baselines.sh --scenarios minimal_specialization --seeds 42 --num-iterations 1` on a real host

Closes #384

### CODE REVIEW

**Comment by rjwalters** [comment]
## Loom Judge review — approved

5th Option-A launcher in this batch (sibling to #393/#394/#395/#399). Pattern is well established; this one cleanly adapts it to the #384 PPO-baseline-training task. CI green (6/6), mergeStateStatus CLEAN, 21/21 new tests pass, 63/63 across sibling launchers (no regressions).

### Three builder-flagged caveats — all blessed

1. **Trainer choice `ippo` is correct.** Verified `run_tier1_cell.py:201-205`:
   ```
   "ippo": TrainerSpec(name="ippo",
       description="Independent PPO baseline (--algorithm ppo, no extras).",
       train_extra=("--algorithm", "ppo"))
   ```
   That is the literal "baseline PPO" arm #371 wants to ship — no centralized critic, no asymmetric init, no value-loss tricks. The #384 issue body's `train.py --seeds 42 43 44` reference was wrong (`train.py` only takes a single `--seed`); routing through `run_tier1_cell.py --trainer ippo` is the right dispatcher and matches #394/#395.

2. **`positional_default-v1` gap is real but non-blocking.** Confirmed: `positional_default` is in `SCENARIO_REGISTRY` (`scenarios_generated.py:663`) but absent from `SCENARIO_VERSIONS` (`registry.py`). Training works because the dispatcher consumes bare names. The runbook correctly flags this as a #371-loader follow-up — not a launcher blocker.

3. **`;` chaining is the right call for a release-training batch.** One failing scenario writing `verdict=no_data` shouldn't take down the other 5 — #371's packager can pick up partial success. Contrast with het_ppo's `&&` chaining, where each scenario is an indivisible Phase. Test `test_dry_run_loops_over_scenarios_with_semicolon` locks the semantics in, and the runbook explicitly documents the tradeoff.

### Other checks

- **Drift check (`test_launcher_trainer_name_matches_driver`)** imports `TRAINERS` from `run_tier1_cell.py` and asserts `'ippo' in TRAINERS` — same pattern that caught the #395/#394 drift earlier today. Will fail loudly if someone renames/removes `ippo`.
- **Launcher pattern adherence**: `--dry-run` skips ssh + side effects; host auto-resolution (PRIMARY → CLUSTER → LAMBDA → GCP); `BatchMode=yes -o ConnectTimeout=5` SSH safety check; bootstrap correctly handles macOS bare PATH, `PYO3_USE_ABI3_FORWARD_COMPATIBILITY=1`, unset `RUSTC_WRAPPER`, and uv-without-pip via `uv pip install pip` (per the MEMORY notes from earlier today on cluster bootstrap order).
- **Runbook quality**: 4 plans (full / sanity / shard / smoke) all distinct and useful. Plan C correctly references cluster reliability ranking (prefer alc-2/6/9, avoid alc-5). Handoff section explicitly hands off to #371.
- **Test discipline**: 21 tests on the high side, but consistent with siblings. Each test pins a real contract (arg parsing, host resolution, driver-cmd shape, session-name compaction, runbook commands, drift check). No shallow filler.
- **Scope discipline**: 3 new files, no edits to drivers, no edits to unrelated code. Closes #384.

Approving.

— Loom Judge
